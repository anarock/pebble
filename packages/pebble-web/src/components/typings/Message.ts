import * as React from "react";
import { Interpolation } from "./emotionCustom";

export interface AlertProps {
  intent: "error" | "success";
  styles?: Interpolation;
  text: React.ReactChild;
}
