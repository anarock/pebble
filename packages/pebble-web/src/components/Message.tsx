import * as React from "react";

import { _colors } from "./Toast";
import { messageWrapper } from "./styles/Message.styles";
import { AlertProps } from "./typings/Message";

const Message: React.FunctionComponent<AlertProps> = ({
  intent,
  styles,
  text
}) => {
  const bColor = _colors[intent];

  return (
    <div
      css={[messageWrapper, styles]}
      style={{
        backgroundColor: bColor
      }}
    >
      <i
        className={`pi ${(intent === "success" && "pi-radio-check-filled") ||
          ""} ${(intent === "error" && "pi-close-circle-filled") || ""}`}
      />
      {text}
    </div>
  );
};

export default Message;
