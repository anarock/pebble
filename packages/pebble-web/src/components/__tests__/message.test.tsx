import renderer from "react-test-renderer";
import Message from "../Message";
import * as React from "react";

describe("Message", () => {
  test("snapshot", () => {
    const mesage = renderer.create(
      <Message intent="success" text="Hello there" />
    );

    const tree = mesage.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
