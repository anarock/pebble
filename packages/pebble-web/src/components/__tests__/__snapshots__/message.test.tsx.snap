// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message snapshot 1`] = `
.emotion-0 {
  border-radius: 3px;
  font-weight: 400;
  color: #ffffff;
  font-size: 14px;
  height: 46px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
}

.emotion-0 > i {
  margin-right: 10px;
}

<div
  className="emotion-0"
  style={
    Object {
      "backgroundColor": "#10AC60",
    }
  }
>
  <i
    className="pi pi-radio-check-filled"
  />
  Hello there
</div>
`;
